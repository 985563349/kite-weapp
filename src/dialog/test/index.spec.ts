import path from 'node:path';
import simulate from 'miniprogram-simulate';

describe('dialog', () => {
  const KDialog = simulate.load(path.resolve(__dirname, '../dialog'), 'k-dialog', {
    rootPath: path.resolve(__dirname, '../../'),
  });

  test('should render dialog and match snapshot', async () => {
    const comp = simulate.render(
      simulate.load({
        usingComponents: {
          'k-dialog': KDialog,
        },
        template: `<k-dialog open />`,
      })
    );
    comp.attach(document.createElement('parent-wrapper'));

    await simulate.sleep(16 * 3); // wait three frames for the animation to complete.
    expect(comp.toJSON()).toMatchSnapshot();
  });
});
