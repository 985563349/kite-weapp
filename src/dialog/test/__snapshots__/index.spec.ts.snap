// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dialog should render dialog and match snapshot 1`] = `
<main>
  <wx-root-portal
    enable="{{false}}"
  >
    <wx-view
      ariaLabel="close"
      ariaRole="button"
      class="k-overlay k-fade-enter-to k-fade-enter-active enter-to-class enter-active-class k-class"
      id=""
      style="-webkit-transition-duration:300ms;transition-duration:300ms;z-index:11000"
      bind:tap="onTap"
      catch:touchmove="noop"
      bind:transitionend="onTransitionEnd"
    />
  </wx-root-portal>
  <wx-root-portal
    enable="{{false}}"
  >
    <wx-view
      class="k-popup k-popup--center k-popup--round k-center-enter-to k-center-enter-active enter-to-class enter-active-class k-class"
      id=""
      style="-webkit-transition-duration:300ms;transition-duration:300ms;z-index:11000;width:320px"
      catch:touchmove="noop"
      bind:transitionend="onTransitionEnd"
    >
      <wx-view
        class="k-dialog-header"
      />
      <wx-view
        class="k-dialog-body"
      />
      <wx-view
        class="k-dialog-footer"
      >
        <wx-button
          appParameter=""
          ariaLabel=""
          class="k-button k-button--primary k-button--ghost k-button--square k-class"
          formType=""
          hoverClass="k-button--active hover-class"
          hoverStartTime="{{20}}"
          hoverStayTime="{{70}}"
          hoverStopPropagation="{{false}}"
          id=""
          lang="en"
          openType=""
          phoneNumberNoQuotaToast="{{false}}"
          sendMessageImg=""
          sendMessagePath=""
          sendMessageTitle=""
          showMessageCard="{{false}}"
          style=""
          bind:agreeprivacyauthorization="agreeprivacyauthorization"
          bind:chooseavatar="chooseavatar"
          bind:contact="contact"
          bind:error="error"
          bind:getphonenumber="getPhoneNumber"
          bind:getuserinfo="getUserInfo"
          bind:launchapp="launchapp"
          bind:opensetting="opensetting"
          bind:tap="onTap"
        >
          Confirm
        </wx-button>
      </wx-view>
    </wx-view>
  </wx-root-portal>
</main>
`;
